Laser diode

amtrang opened the pull request.

amtrang committed:
Laser Diode Prototype 1

Started working on the laser diode which uses a collider trigger to hit the player.

amtrang committed:
Laser Diode Prototype 2

Added flicker option so that the laser can turn itself on and off at intervals.

amtrang committed:
Laser Diode Prototype Finished

The laser diode extends to a max distance that can be set. Also, the laser stops at objects (except the player). You can try moving the wall in the scene manager the laser hits when playing the game.

CristionD reviewed (commented):
I request 3 modifications:
1. Place the `LaserDiode` script on the 'Laser Diode' gameobject rather than 'Fire Point'.
2. Add a variable that determines the hit width of the laser (utilize `Physics.SphereCast` when the width is greater than 0).
3. Ensure the line renderer follows the laser direction (utilizing the direction from the 1st position to the 2nd position of the renderer component for the raycast is an option).

amtrang committed:
Raycast & Line Renderer Misalign

At certain angles, the raycast and line renderer are misaligned. This should be fixed soon.

CristionD reviewed (commented):
Ignore this.

CristionD committed:
Remedy laser direction bug

amtrang committed:
SphereCast and Code Organization 1

Implemented SphereCast instead of RayCast so that the laser can hit at a wider area. Also made some adjustments to the LaserDiode code like when the laser is flickering, before it turns back on, it uses the SphereCast before turning on the line renderer so that it doesn't show itself at the wrong position in a single frame.

CristionD committed:
Merge branch 'main' into laser_diode

CristionD committed:
Refactor `LaserDiode` script, prefab, and testing scene.
